cmd/api-linter/cli.go

```diff
@@ -105,13 +105,18 @@ func runCLI(rules lint.Rules, configs lint.Configs, args []string) error {
 				case "json":
 					marshal = json.Marshal
 				}
-				b, err := marshal(problems)
-				if err != nil {
-					return err
-				}
 
-				if _, err = w.Write(b); err != nil {
-					return err
+				if c.String("fmt") == "summary" {
+					summary := createSummary(lintResponses)
```

Comment:
Actually, it would be super cool if you implemented Marshal and then the switch statement would be super clean. 

```
case "summary":
  marshal = summary.Marshal

Reply:
added an inline implementation of Marshal. Will that suffice?